core/src/main/scala/cats/data/ContT.scala

```diff
@@ -188,6 +190,38 @@ object ContT {
   def later[M[_], A, B](fn: => (B => M[A]) => M[A]): ContT[M, A, B] =
     DeferCont(() => FromFn(AndThen(fn)))
 
+  /*
+   * Limits the continuation of any inner [[shiftT]]
+   */
+  def resetT[M[_]: Monad: Defer, R, R2](contT: ContT[M, R, R]): ContT[M, R2, R] =
+    ContT.liftF(contT.eval)
+
+  /*
+   * Captures the continuation up to the nearest enclosing [[resetT]] and passes
+   * it to f
+   *
+   * {{{
+   *   for {
+   *     _ <- ContT.resetT(
+   *            for {
+   *              _ <- ContT.liftF(IO.println("1"))
+   *              _ <- ContT.shiftT{ (k: Unit => IO[Unit])
+   *                     for {
+   *                       _ <- ContT.liftF(IO.println("2"))
+   *                       _ <- ContT.liftF(k(()))
+   *                       _ <- ContT.liftF(IO.println("4"))
+   *                     }
+   *                   }
+   *              _ <- ContT.liftF(IO.println("3"))
+   *            } yield ()
+   *         )
+   *     _ <- ContT.liftF(IO.println("5"))
+   *   } yield ()
+   * }}}
```

Comment:
can you add what this is supposed to do? It you don't understand shift and reset (and I am foggy on it), I don't think this comment helps you.

Reply:
Yup 👍